clients/cli/src/main/java/org/apache/gravitino/cli/outputs/PlainFormat.java

```diff
@@ -21,71 +21,247 @@
+      if (metalakes == null || metalakes.length == 0) {
+        return null;
       } else {
         List<String> metalakeNames =
             Arrays.stream(metalakes).map(Metalake::name).collect(Collectors.toList());
-        String all = String.join(System.lineSeparator(), metalakeNames);
-        System.out.println(all);
+        return NEWLINE_JOINER.join(metalakeNames);
       }
     }
   }
 
-  static final class CatalogPlainFormat implements OutputFormat<Catalog> {
+  /**
+   * Formats a single {@link Catalog} instance as a comma-separated string. Output format: name,
+   * type, provider, comment
+   */
+  static final class CatalogPlainFormat extends PlainFormat<Catalog> {
+    public CatalogPlainFormat(CommandContext context) {
+      super(context);
+    }
+
+    /** {@inheritDoc} */
     @Override
-    public void output(Catalog catalog) {
-      System.out.println(
-          catalog.name()
-              + ","
-              + catalog.type()
-              + ","
-              + catalog.provider()
-              + ","
-              + catalog.comment());
+    public String getOutput(Catalog catalog) {
+      return COMMA_JOINER.join(
+          catalog.name(), catalog.type(), catalog.provider(), catalog.comment());
     }
   }
 
-  static final class CatalogsPlainFormat implements OutputFormat<Catalog[]> {
+  /**
+   * Formats an array of Catalogs, outputting one name per line. Returns null if the array is empty
+   * or null.
+   */
+  static final class CatalogsPlainFormat extends PlainFormat<Catalog[]> {
+    public CatalogsPlainFormat(CommandContext context) {
+      super(context);
+    }
+
+    /** {@inheritDoc} */
     @Override
-    public void output(Catalog[] catalogs) {
-      if (catalogs.length == 0) {
-        System.out.println("No catalogs exist.");
+    public String getOutput(Catalog[] catalogs) {
+      if (catalogs == null || catalogs.length == 0) {
+        output("No catalogs exists.", System.err);
+        return null;
       } else {
         List<String> catalogNames =
             Arrays.stream(catalogs).map(Catalog::name).collect(Collectors.toList());
-        String all = String.join(System.lineSeparator(), catalogNames);
-        System.out.println(all);
+        return NEWLINE_JOINER.join(catalogNames);
       }
     }
   }
+
+  /**
+   * Formats a single {@link Schema} instance as a comma-separated string. Output format: name,
+   * comment
+   */
+  static final class SchemaPlainFormat extends PlainFormat<Schema> {
+    public SchemaPlainFormat(CommandContext context) {
+      super(context);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getOutput(Schema schema) {
+      return COMMA_JOINER.join(schema.name(), schema.comment());
+    }
+  }
+
+  /**
+   * Formats an array of Schemas, outputting one name per line. Returns null if the array is empty
+   * or null.
+   */
+  static final class SchemasPlainFormat extends PlainFormat<Schema[]> {
```

Comment:
The difference between `SchemaPlainFormat` and `SchemasPlainFormat` can be made more noticeable. e.g. `SchemaListPlainFormat`.